pkg/agent/querier/querier.go

```diff
@@ -47,6 +48,7 @@ type AgentQuerier interface {
 	GetMemberlistCluster() memberlist.Interface
 	GetNodeLister() corelisters.NodeLister
 	GetBGPPolicyInfoQuerier() querier.AgentBGPPolicyInfoQuerier
+	GetFqdnCache(querier.FQDNCacheFilter) []types.DnsCacheEntry
```

Comment:
This seems wrong: there is no need to add this function to the interface because you can already call `GetNetworkPolicyInfoQuerier().GetFqdnCache(filter)`, which is what we do for other handlers.